crates/iroha_cli/src/main.rs

```diff
@@ -1000,13 +1000,160 @@ mod asset {
             Ok(())
         }
     }
+}
+
+mod nft {
+    use super::*;
+
+    /// Subcommand for dealing with NFT
+    #[derive(clap::Subcommand, Debug)]
+    pub enum Args {
+        /// Register NFT
+        Register(Register),
+        /// Unregister NFT
+        Unregister(Unregister),
+        /// Transfer NFT between accounts
+        Transfer(Transfer),
+        /// Get info of NFT
+        Get(Get),
+        /// List NFTs
+        #[clap(subcommand)]
+        List(List),
+        /// Get a value from NFT
+        GetKeyValue(GetKeyValue),
+        /// Set a key-value entry in NFT
+        SetKeyValue(SetKeyValue),
+        /// Remove a key-value entry from NFT
+        RemoveKeyValue(RemoveKeyValue),
+    }
+
+    impl RunArgs for Args {
+        fn run(self, context: &mut dyn RunContext) -> Result<()> {
+            use self::Args::*;
+            match_all!(
+                (self, context),
+                { Register, Unregister, Transfer, Get, List, GetKeyValue, SetKeyValue, RemoveKeyValue }
+            )
+        }
+    }
+
+    /// Register subcommand for NFT
+    #[derive(Debug, clap::Args)]
+    pub struct Register {
+        /// NFT id (in form `name$domain_name`)
+        #[arg(long)]
+        pub id: NftId,
+        #[command(flatten)]
+        pub metadata: MetadataArgs,
+    }
+
+    impl RunArgs for Register {
+        fn run(self, context: &mut dyn RunContext) -> Result<()> {
+            let Self { id, metadata } = self;
+            let create_nft = iroha::data_model::isi::Register::nft(Nft::new(id, metadata.load()?));
+            submit([create_nft], Metadata::default(), context).wrap_err("Failed to register NFT")
+        }
+    }
+
+    /// Unregister subcommand for NFT
+    #[derive(Debug, clap::Args)]
+    pub struct Unregister {
+        /// NFT id (in form `name$domain_name`)
+        #[arg(long)]
+        pub id: NftId,
+    }
+
+    impl RunArgs for Unregister {
+        fn run(self, context: &mut dyn RunContext) -> Result<()> {
+            let unregister_nft = iroha::data_model::isi::Unregister::nft(self.id);
+            submit([unregister_nft], Metadata::default(), context)
+                .wrap_err("Failed to unregister NFT")
+        }
+    }
+
+    /// Transfer NFT between accounts
+    #[derive(clap::Args, Debug)]
+    pub struct Transfer {
+        /// NFT id to transfer (in form `name$domain_name`)
+        #[arg(long)]
+        pub id: NftId,
+        /// Account from which to transfer (in form `name@domain_name`)
+        #[arg(short, long)]
+        pub from: AccountId,
```

Comment:
What do you think about deriving the source account from the queried object owner to eliminate the need for user input?

Reply:
I think it is a good idea, but if we change it for NFT, it is better to change it for domains also so CLI will be consistent. Opened #5312